scripts/footprint/size_report

```diff
@@ -495,89 +476,101 @@ def set_root_path_for_unmapped_symbols(symbol_dict, addr_range, processed):
             if cur is None:
                 cur = part
             else:
                 cur = str(Path(cur, part))
-            if cur in tree:
-                tree[cur] += size
-            else:
-                tree[cur] = size
 
-    total_size = 0
-    nodes = {}
-    nodes[':'] = 0
+            results = findall_by_attr(root, cur, name="identifier")
+            if results:
+                item = results[0]
+                item.size += size
+                parent = item
+            else:
+                if node:
+                    parent = node
+                node = TreeNode(name=str(part), identifier=cur, size=size, parent=parent)
 
     for name, sym in symbol_dict.items():
         for symbol in sym:
             size = get_symbol_size(symbol['symbol'])
             for file in symbol['mapped_files']:
                 path = Path(file, name)
-                _insert_one_elem(nodes, path, size)
+                if path.is_absolute():
+                    zb = Path(args.zephyrbase)
+                    if zb.parent in path.parents:
+                        path = path.relative_to(zb.parent)
 
-    ret = {'nodes': nodes,
-           'size': total_size}
-    return ret
+                _insert_one_elem(root, path, size)
 
+    return root
 
-def print_tree(data, total, depth):
-    base = args.zephyrbase
-    totp = 0
-
-    bcolors_ansi = {
-        "HEADER"    : '\033[95m',
-        "OKBLUE"    : '\033[94m',
-        "OKGREEN"   : '\033[92m',
-        "WARNING"   : '\033[93m',
-        "FAIL"      : '\033[91m',
-        "ENDC"      : '\033[0m',
-        "BOLD"      : '\033[1m',
-        "UNDERLINE" : '\033[4m'
-    }
-    if platform.system() == "Windows" or args.nocolor:
-        # Set all color codes to empty string on Windows
-        #
-        # TODO: Use an approach like the pip package 'colorama' to
-        # support colors on Windows
-        bcolors = dict.fromkeys(bcolors_ansi, '')
-    else:
-        bcolors = bcolors_ansi
+def print_any_tree(root, total_size):
 
-    print('{:92s} {:10s} {:8s}'.format(
-        bcolors["FAIL"] + "Path", "Size", "%" + bcolors["ENDC"]))
+    print('{:101s} {:7s} {:8s}'.format(
+        Fore.YELLOW + "Path", "Size", "%" + Fore.RESET))
     print('=' * 110)
-    for i in sorted(data):
-        p = i.split(os.path.sep)
-        if depth and len(p) > depth:
-            continue
+    for row in RenderTree(root):
+        f = len(row.pre) + len(row.node.name)
+        s = str(row.node.size).rjust(100-f)
+        percent = 100 * float(row.node.size) / float(total_size)
+
+        cc = cr = ""
+        if not row.node.children:
+            cc = Fore.CYAN
+            cr = Fore.RESET
+        elif row.node.name.endswith(".c") or row.node.name.endswith(".h"):
+            cc = Fore.GREEN
+            cr = Fore.RESET
+
+        print(f"{row.pre}{cc}{row.node.name}{cr} {s} {Fore.BLUE}{percent:.2f}%{Fore.RESET}")
+    print('=' * 110)
+    print(f'{total_size:>100}')
```

Comment:
Probably needs to be `>101`. It is misaligned when I ran it.

Reply:
fixed